test/js/metrics-api.test.js

```diff
@@ -99,6 +99,49 @@ describe('Library for client-side usage analytics', () => {
         )
       })
     )
+    process.nextTick(() => {
+      global.fetch.mockClear();
+      done()
+    })
+  })
+  it('logs classList and id when link is clicked', done => {
+    // Spy on `fetch()` and its contingent methods like `json()`,
+    // because we want to intercept the outgoing request
+    const mockSuccessResponse = {}
+    const mockJsonPromise = Promise.resolve(mockSuccessResponse)
+    const mockFetchPromise = Promise.resolve({
+      json: () => {
+        mockJsonPromise
+      }
+    })
+    jest.spyOn(global, 'fetch').mockImplementation(() => {
+      mockFetchPromise
+    })
+
+    // Mock the user context, to mimic auth'd state
+    const userContext = { accessToken: 'test' }
+    jest.spyOn(UserProvider, 'useContextUser')
+      .mockImplementation(() => {
+        return userContext
+      })
```

Comment:
I think this block could be omitted without issue, given we're not testing auth-related code.